packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -123,56 +265,85 @@ class UserSessionManager implements IUserSessionManager {
     }
   }
 
+  /**
+   * Set user Id
+   * @param userId
+   */
   setUserId(userId?: Nullable<string>) {
     state.session.rl_user_id.value = userId;
-    this.storage?.set('rl_user_id', userId);
   }
 
-  // TODO: should we reset traits in value is null?
+  /**
+   * Set user traits
+   * @param userId
+   */
   setUserTraits(traits?: Nullable<ApiObject>) {
     if (traits) {
       state.session.rl_trait.value = mergeDeepRight(state.session.rl_trait.value || {}, traits);
-      window.setTimeout(() => {
-        this.storage?.set('rl_trait', state.session.rl_trait.value);
-      }, 1);
     }
   }
 
+  /**
+   * Set group Id
+   * @param userId
+   */
   setGroupId(groupId?: Nullable<string>) {
     state.session.rl_group_id.value = groupId;
-    this.storage?.set('rl_group_id', groupId);
   }
 
-  // TODO: should we reset traits in value is null?
-  setGroupTraits(traits?: Nullable<ApiOptions>) {
+  /**
+   * Set group traits
+   * @param userId
+   */
+  setGroupTraits(traits?: Nullable<ApiObject>) {
     if (traits) {
       state.session.rl_group_trait.value = mergeDeepRight(
         state.session.rl_group_trait.value || {},
         traits,
       );
-      window.setTimeout(() => {
-        this.storage?.set('rl_group_trait', state.session.rl_group_trait.value);
-      }, 1);
     }
   }
 
+  /**
+   * Set initial referrer
+   * @param userId
+   */
+  setInitialReferrer(referrer?: string) {
+    state.session.rl_page_init_referrer.value = referrer;
+  }
+
+  /**
+   * Set initial referring domain
+   * @param userId
+   */
+  setInitialReferringDomain(referrer?: string) {
+    state.session.rl_page_init_referring_domain.value = referrer;
```

Comment:
should we use persistedSessionStorageKeys fro all key names like rl_page_init_referring_domain in session?

Reply:
Ideally, I'd separate the corresponding property names used internally and in storage. We're anyway tying those two sets of fields in the effect blocks.

`state.session.initialReferringDomain`
`state.session.userId`
etc.

Reply:
Addressed